biztown/src/components/FormInputLayout.js

```diff
@@ -1,260 +1,428 @@
+      aboutSectionAnswerObj.otherGrade = null;
     }
+    // remove school district from response
+    aboutSectionAnswerObj.schoolDistrict = null;
+
+    // format free response data
+    var depositFormData = this.state.freeResponseData.depositForm
+      .rowEntries;
+    var depositFormAnswerObj = {
+      q11_answer: `${depositFormData.dollarAmount_5}.${depositFormData.centAmount_5}`,
+      q12_answer: `${depositFormData.dollarAmount_6}.${depositFormData.centAmount_6}`,
+      q13_answer: `${depositFormData.dollarAmount_7}.${depositFormData.centAmount_7}`,
+    };
+
+    var registerFormData = this.state.freeResponseData.registerEntries
+      .rowEntries;
 
-    handleFreeResponseSectionChange = (bankFormType, event) => {
-        const {name, value} = event.target
-        if (bankFormType === "checkSlip") {
-            this.setState(prevState => {
-                var freeResponseData = JSON.parse(JSON.stringify(prevState.freeResponseData));
-                freeResponseData[bankFormType][name] = value;
-                return { freeResponseData };
-            })
-        } else {
-            this.setState(prevState => {
-                var freeResponseData = JSON.parse(JSON.stringify(prevState.freeResponseData));
-                freeResponseData[bankFormType]["rowEntries"][name] = value;
-                return { freeResponseData };
-            })
+    var registerFormAnswerObj = {
+      q17_answer: `${registerFormData.balanceDollarAmount_0}.${registerFormData.balanceCentAmount_0}`,
+      q18_answer: `${registerFormData.balanceDollarAmount_1}.${registerFormData.balanceCentAmount_1}`,
+      q19_answer: registerFormData.transactionDesc_2,
+      q20_answer: `${registerFormData.paymentDollarAmount_2}.${registerFormData.paymentCentAmount_2}`,
+      q21_answer: `${registerFormData.balanceDollarAmount_2}.${registerFormData.balanceCentAmount_2}`,
+      q22_answer: `${registerFormData.balanceDollarAmount_3}.${registerFormData.balanceCentAmount_3}`,
+    };
+
+    // construct completed form from different objects
+    var completedForm = {
+      ...depositFormAnswerObj,
+      ...registerFormAnswerObj,
+      ...aboutSectionAnswerObj,
+      ...this.state.multipleChoiceData,
+      ...this.state.freeResponseData.checkSlip,
+      ...this.state.personalFinanceData,
+    };
+
+    console.log("completedForm", completedForm);
+
+    var submitPath = this.props.postTest
+      ? "postassessment/"
+      : "preassessment/";
+    DataManager.post(submitPath, completedForm).then(() => {
+      this.props.history.push("/completionpage");
+    });
+  };
+
+  validatePage = (pageObj) => {
+    for (let question of pageObj.keys) {
+      const element = document.getElementById(`${question}`);
+
+      if (document.getElementById(`${question}`).type === "radio") {
+        const section = this.state[`${pageObj.title}`];
+
+        if (!(element.name in section)) {
+          alert(`Please choose an option for ${question}.`);
```

Comment:
One thing that you could do, which is a lot more work is pass in an error prop each component s.t. if values do not have data, they have the error styling to be consistent which can be resolved on successful completion of the page.

Reply:
Added to backlog.